Fix TestBPF

tklauser opened the pull request:
Fixes #22078

See commits for details.

tklauser committed:
Revert "Revert "bpf_test: Skip instead of Fatal TestBPF when ...""

This reverts commit 8d35158ba276 ("Revert "bpf_test: Skip instead of Fatal TestBPF when ..."").

Reason for revert:

This breaks TestBPF in DatapathPrivilegedUnitTests.

A follow-up patch will fix the error the original revert commit was
trying to solve:

bpf_test.go:149: new coll: program test_ct4_rst1_check: load program: invalid argument: nonzero insn_off 25 for the first func info recordprocessed 0 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0

For #22078

Signed-off-by: Tobias Klauser <tobias@cilium.io>

pchaigno commented on .github/workflows/lint-bpf-checks.yaml:
```diff
@@ -133,7 +133,7 @@ jobs:
           fetch-depth: 0
       - name: Run BPF tests with code coverage reporting
         run: |
-          make -C test run_bpf_tests COVER=1 || (echo "Run 'make -C test run_bpf_tests COVER=1' locally to investigate failures"; exit 1)
```
Do we need an issue to track this change and remember to revert once the underlying issue is fixed?

tklauser replied:
Thanks, good point. I've created https://github.com/cilium/cilium/issues/22088 and cross-referenced in a comment in the line above.

pchaigno commented on .github/workflows/lint-bpf-checks.yaml:
```diff
@@ -133,7 +133,8 @@ jobs:
           fetch-depth: 0
       - name: Run BPF tests with code coverage reporting
```
This title isn't accurate anymore :sweat_smile: Not sure if it's worth fixing.

tklauser replied:
🤦 I've fixed and re-pushed.

tklauser committed:
.github/workflows: disable coverage in BPF tests

This needs to be fixed in coverbee first, see
https://github.com/cilium/cilium/pull/22043#issuecomment-1307887505

For #22078
For #22088

Suggested-by: Timo Beckers <timo@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>
